src/redprl/refiner_types.fun

```diff
@@ -246,42 +244,42 @@ struct
     fun EqElim alpha jdg =
       let
         val _ = RedPrlLog.trace "WBool.EqElim"
-        val H >> AJ.EQ ((if0, if1), (ty, l, k)) = jdg
+        val H >> AJ.EQ ((if0, if1), (ty, l)) = jdg
         (* if(FCOM) steps to COM *)
-        val k = K.meet (k, K.COM)
+        val k = K.COM
         val Syn.WIF ((x, c0x), m0, (t0, f0)) = Syn.out if0
         val Syn.WIF ((y, c1y), m1, (t1, f1)) = Syn.out if1
 
         (* motive *)
         val z = alpha 0
         val c0z = VarKit.rename (z, x) c0x
         val c1z = VarKit.rename (z, y) c1y
-        val Hz = H @> (z, AJ.TRUE (Syn.into Syn.WBOOL, inherentLevel, inherentKind))
-        val goalTy = makeEqType (Hz) ((c0z, c1z), NONE, K.top)
+        val Hz = H @> (z, AJ.TRUE (Syn.into Syn.WBOOL, inherentLevel))
+        val goalTy = makeEqType Hz ((c0z, c1z), l, k)
```

Comment:
Reverted to safer checking. @cangiuli The question is, for `wool-rec` to be well-typed, maybe the motive `z.C` does not need to be Kan everywhere, but only at `C[m/x]` where `m` is the eliminated term. Did you prove or disprove this?